diff --git a/Cargo.lock b/Cargo.lock
index 23d4d90..604e1be 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1033,7 +1033,10 @@ dependencies = [
 name = "cosmic-ext-applet-sysinfo"
 version = "0.1.0"
 dependencies = [
+ "i18n-embed",
+ "i18n-embed-fl",
  "libcosmic",
+ "rust-embed",
  "sysinfo",
  "tracing",
  "tracing-subscriber",
@@ -4270,9 +4273,9 @@ checksum = "6c20b6793b5c2fa6553b250154b78d6d0db37e72700ae35fad9387a46f487c97"
 
 [[package]]
 name = "rust-embed"
-version = "8.7.2"
+version = "8.11.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "025908b8682a26ba8d12f6f2d66b987584a4a87bc024abc5bbc12553a8cd178a"
+checksum = "04113cb9355a377d83f06ef1f0a45b8ab8cd7d8b1288160717d66df5c7988d27"
 dependencies = [
  "rust-embed-impl",
  "rust-embed-utils",
@@ -4281,9 +4284,9 @@ dependencies = [
 
 [[package]]
 name = "rust-embed-impl"
-version = "8.7.2"
+version = "8.11.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "6065f1a4392b71819ec1ea1df1120673418bf386f50de1d6f54204d836d4349c"
+checksum = "da0902e4c7c8e997159ab384e6d0fc91c221375f6894346ae107f47dd0f3ccaa"
 dependencies = [
  "proc-macro2",
  "quote",
@@ -4294,9 +4297,9 @@ dependencies = [
 
 [[package]]
 name = "rust-embed-utils"
-version = "8.7.2"
+version = "8.11.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "f6cc0c81648b20b70c491ff8cce00c1c3b223bb8ed2b5d41f0e54c6c4c0a3594"
+checksum = "5bcdef0be6fe7f6fa333b1073c949729274b05f123a0ad7efcb8efd878e5c3b1"
 dependencies = [
  "sha2",
  "walkdir",
diff --git a/Cargo.toml b/Cargo.toml
index b36f14d..f2627f6 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,9 +11,16 @@ repository = "https://github.com/cosmic-utils/cosmic-ext-applet-sysinfo.git"
 lto = "fat"
 
 [dependencies]
+i18n-embed = { version = "0.16.0", default-features = false, features = [
+  "rust-embed",
+  "fluent-system",
+  "desktop-requester",
+] }
+i18n-embed-fl = { version = "0.10.0", default-features = false }
 libcosmic = { git = "https://github.com/pop-os/libcosmic", default-features = false, features = [
 	"applet",
 ] }
+rust-embed = { version = "8.11.0", default-features = false }
 sysinfo = { version = "0.37.0", default-features = false, features = [
 	"system",
 	"network",
diff --git a/i18n.toml b/i18n.toml
new file mode 100644
index 0000000..76f7c31
--- /dev/null
+++ b/i18n.toml
@@ -0,0 +1,4 @@
+fallback_language = "en"
+
+[fluent]
+assets_dir = "i18n"
diff --git a/i18n/en/cosmic_ext_applet_sysinfo.ftl b/i18n/en/cosmic_ext_applet_sysinfo.ftl
new file mode 100644
index 0000000..3543021
--- /dev/null
+++ b/i18n/en/cosmic_ext_applet_sysinfo.ftl
@@ -0,0 +1 @@
+include-swap-in-ram-toggle = Include swap in RAM
diff --git a/src/applet.rs b/src/applet.rs
index 50dc3c8..b3d966e 100644
--- a/src/applet.rs
+++ b/src/applet.rs
@@ -5,16 +5,22 @@ use std::{
 };
 
 use sysinfo::{CpuRefreshKind, MemoryRefreshKind, Networks, RefreshKind, System};
+use tracing::{debug, trace};
 
-use crate::config::{APP_ID, SysInfoConfig};
+use crate::{
+    config::{APP_ID, Flags, SysInfoConfig},
+    fl,
+};
 
 pub fn run() -> cosmic::iced::Result {
-    cosmic::applet::run::<SysInfo>(())
+    cosmic::applet::run::<SysInfo>(Flags::new())
 }
 
 struct SysInfo {
     core: cosmic::app::Core,
+    popup: Option<cosmic::iced::window::Id>,
     config: SysInfoConfig,
+    config_handler: Option<cosmic::cosmic_config::Config>,
     system: System,
     networks: Networks,
     cpu_usage: f32,
@@ -68,7 +74,12 @@ impl SysInfo {
         );
 
         self.cpu_usage = self.system.global_cpu_usage();
-        self.ram_usage = (self.system.used_memory() * 100) / self.system.total_memory();
+        self.ram_usage = if self.config.include_swap_in_ram {
+            ((self.system.used_memory() + self.system.used_swap()) * 100)
+                / (self.system.total_memory() + self.system.total_swap())
+        } else {
+            (self.system.used_memory() * 100) / self.system.total_memory()
+        };
 
         self.networks.refresh(true);
 
@@ -87,13 +98,16 @@ impl SysInfo {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Message {
     Tick,
+    ToggleWindow,
+    PopupClosed(cosmic::iced::window::Id),
+    ToggleIncludeSwapInRam(bool),
 }
 
 impl cosmic::Application for SysInfo {
-    type Flags = ();
+    type Flags = Flags;
     type Message = Message;
     type Executor = cosmic::SingleThreadExecutor;
 
@@ -101,13 +115,18 @@ impl cosmic::Application for SysInfo {
 
     fn init(
         core: cosmic::app::Core,
-        _flags: Self::Flags,
+        flags: Self::Flags,
     ) -> (Self, cosmic::app::Task<Self::Message>) {
-        let config = SysInfoConfig::config();
+        let config = flags.config;
 
+        let memory_config = if config.include_swap_in_ram {
+            MemoryRefreshKind::nothing().with_ram().with_swap()
+        } else {
+            MemoryRefreshKind::nothing().with_ram()
+        };
         let system = System::new_with_specifics(
             RefreshKind::nothing()
-                .with_memory(MemoryRefreshKind::nothing().with_ram())
+                .with_memory(memory_config)
                 .with_cpu(CpuRefreshKind::nothing().with_cpu_usage()),
         );
         let networks = Networks::new_with_refreshed_list();
@@ -118,7 +137,9 @@ impl cosmic::Application for SysInfo {
         (
             Self {
                 core,
+                popup: None,
                 config,
+                config_handler: flags.config_handler,
                 system,
                 networks,
                 cpu_usage: 0.0,
@@ -144,9 +165,55 @@ impl cosmic::Application for SysInfo {
         cosmic::iced::time::every(Duration::from_secs(1)).map(|_| Message::Tick)
     }
 
+    fn on_close_requested(&self, id: cosmic::iced::window::Id) -> Option<Message> {
+        Some(Message::PopupClosed(id))
+    }
+
     fn update(&mut self, message: Message) -> cosmic::app::Task<Self::Message> {
+        match message {
+            // don't spam the logs with the tick
+            Message::Tick => trace!(?message),
+            _ => debug!(?message),
+        }
+
         match message {
             Message::Tick => self.update_sysinfo_data(),
+            Message::ToggleWindow => {
+                if let Some(id) = self.popup.take() {
+                    debug!("have popup with id={id}, destroying");
+
+                    return cosmic::iced::platform_specific::shell::commands::popup::destroy_popup(
+                        id,
+                    );
+                } else {
+                    debug!("do not have a popup, creating");
+
+                    let new_id = cosmic::iced::window::Id::unique();
+                    debug_assert_eq!(self.popup.replace(new_id), None);
+
+                    let popup_settings = self.core.applet.get_popup_settings(
+                        self.core.main_window_id().unwrap(),
+                        new_id,
+                        None,
+                        None,
+                        None,
+                    );
+
+                    return cosmic::iced::platform_specific::shell::commands::popup::get_popup(
+                        popup_settings,
+                    );
+                }
+            }
+            Message::PopupClosed(id) => {
+                self.popup.take_if(|i| *i == id);
+            }
+            Message::ToggleIncludeSwapInRam(value) => {
+                if let Some(handler) = &self.config_handler
+                    && let Err(error) = self.config.set_include_swap_in_ram(handler, value)
+                {
+                    tracing::error!("{error}")
+                }
+            }
         }
 
         cosmic::task::none()
@@ -163,8 +230,30 @@ impl cosmic::Application for SysInfo {
             .spacing(4)
         };
 
-        let button = cosmic::widget::button::custom(data).class(cosmic::theme::Button::AppletIcon);
+        let button = cosmic::widget::button::custom(data)
+            .class(cosmic::theme::Button::AppletIcon)
+            .on_press_down(Message::ToggleWindow);
 
         cosmic::widget::autosize::autosize(button, cosmic::widget::Id::unique()).into()
     }
+
+    fn view_window(&self, _id: cosmic::iced::window::Id) -> cosmic::Element<'_, Message> {
+        let include_swap_in_ram_toggler = cosmic::iced_widget::row![
+            cosmic::widget::text(fl!("include-swap-in-ram-toggle")),
+            cosmic::widget::Space::with_width(cosmic::iced::Length::Fill),
+            cosmic::widget::toggler(self.config.include_swap_in_ram)
+                .on_toggle(Message::ToggleIncludeSwapInRam),
+        ];
+
+        let data = cosmic::iced_widget::column![
+            // padding comment to make formatting nicer
+            cosmic::applet::padded_control(include_swap_in_ram_toggler)
+        ]
+        .padding([16, 0]);
+
+        self.core
+            .applet
+            .popup_container(cosmic::widget::container(data))
+            .into()
+    }
 }
diff --git a/src/config.rs b/src/config.rs
index 3ed0734..8018640 100644
--- a/src/config.rs
+++ b/src/config.rs
@@ -6,10 +6,12 @@ const CONFIG_VERSION: u64 = 1;
 
 pub const APP_ID: &str = "io.github.cosmic-utils.cosmic-ext-applet-sysinfo";
 
-#[derive(Default, Debug, CosmicConfigEntry)]
+#[derive(Default, Debug, Clone, CosmicConfigEntry)]
 pub struct SysInfoConfig {
     pub include_interfaces: Option<Vec<String>>,
     pub exclude_interfaces: Option<Vec<String>>,
+    /// Whether to include Swap usage in the RAM segment
+    pub(crate) include_swap_in_ram: bool,
 }
 
 impl SysInfoConfig {
@@ -28,3 +30,18 @@ impl SysInfoConfig {
         }
     }
 }
+
+#[derive(Clone, Debug)]
+pub(crate) struct Flags {
+    pub(crate) config: SysInfoConfig,
+    pub(crate) config_handler: Option<cosmic_config::Config>,
+}
+
+impl Flags {
+    pub(crate) fn new() -> Self {
+        Self {
+            config: SysInfoConfig::config(),
+            config_handler: SysInfoConfig::config_handler(),
+        }
+    }
+}
diff --git a/src/i18n.rs b/src/i18n.rs
new file mode 100644
index 0000000..5070d81
--- /dev/null
+++ b/src/i18n.rs
@@ -0,0 +1,44 @@
+use std::sync::LazyLock;
+
+use i18n_embed::{
+    DefaultLocalizer, LanguageLoader, Localizer,
+    fluent::{FluentLanguageLoader, fluent_language_loader},
+    unic_langid::LanguageIdentifier,
+};
+use rust_embed::RustEmbed;
+
+pub fn init(requested_languages: &[LanguageIdentifier]) {
+    if let Err(why) = localizer().select(requested_languages) {
+        tracing::info!("error while loading fluent localizations: {why}");
+    }
+}
+
+#[must_use]
+pub fn localizer() -> Box<dyn Localizer> {
+    Box::from(DefaultLocalizer::new(&*LANGUAGE_LOADER, &Localizations))
+}
+
+#[derive(RustEmbed)]
+#[folder = "i18n/"]
+struct Localizations;
+
+pub static LANGUAGE_LOADER: LazyLock<FluentLanguageLoader> = LazyLock::new(|| {
+    let loader: FluentLanguageLoader = fluent_language_loader!();
+
+    loader
+        .load_fallback_language(&Localizations)
+        .expect("Error whilst loading fallback language");
+
+    loader
+});
+
+#[macro_export]
+macro_rules! fl {
+    ($message_id:literal) => {{
+        i18n_embed_fl::fl!($crate::i18n::LANGUAGE_LOADER, $message_id)
+    }};
+
+    ($message_id:literal, $($args:expr),*) => {{
+        i18n_embed_fl::fl!($crate::i18n::LANGUAGE_LOADER, $message_id, $($args), *)
+    }};
+}
diff --git a/src/main.rs b/src/main.rs
index e5b52bf..0e96ca9 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,5 +1,6 @@
 pub mod applet;
 pub mod config;
+pub(crate) mod i18n;
 
 fn main() -> cosmic::iced::Result {
     // Initialize logging
@@ -9,5 +10,8 @@ fn main() -> cosmic::iced::Result {
 
     tracing::info!("Starting sysinfo applet");
 
+    let requested_languages = i18n_embed::DesktopLanguageRequester::requested_languages();
+    i18n::init(&requested_languages);
+
     applet::run()
 }
